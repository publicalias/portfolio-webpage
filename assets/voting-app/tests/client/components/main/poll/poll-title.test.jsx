"use strict";

//local imports

const PollTitle = require("../../../../../scripts/client/components/main/poll/poll-title");

const { initTestPoll, testWrapper } = require("../../../test-helpers");

//global imports

const { newUser } = require("redux/schemas");
const { initTestEvent, initTestSnapshot } = require("redux/tests/client-tests");
const { reactTests } = require("redux/tests/react-tests");

//utilities

const { testShallow } = testWrapper(PollTitle);

const testForm = initTestPoll(testShallow, "form");
const testView = initTestPoll(testShallow, "view");

//setup

beforeAll(reactTests.setup);
beforeEach(reactTests.inject(PollTitle));

//poll title

describe("PollTitle (form)", () => {

  const testSnapshot = initTestSnapshot(testForm);

  const testBlur = initTestEvent(testForm, "blur", { target: { value: "Title A" } });

  it("should match snapshot (default)", () => testSnapshot());

  it("should match snapshot (author)", () => testSnapshot({ user: newUser({ name: "Author A" }) }));

  it("should call handleBlur on blur", () => testBlur(
    ".qa-title-input",
    [],
    ["formSetTitle", ["Title A"]],
    ["formCheckTitle", ["Title A"]]
  ));

});

describe("PollTitle (view)", () => {

  const testSnapshot = initTestSnapshot(testView);

  it("should match snapshot (default)", () => testSnapshot());

  it("should match snapshot (title)", () => testSnapshot(null, { title: "Title A" }));

  it("should match snapshot (author)", () => testSnapshot(null, { author: "Author A" }));

});
