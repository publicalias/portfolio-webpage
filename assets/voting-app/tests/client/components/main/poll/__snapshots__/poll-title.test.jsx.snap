// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PollTitle (form) should match snapshot (author) 1`] = `
<div
  className="u-align-center"
>
  <input
    className="c-poll-display__input qa-title-input u-margin-half"
    maxLength="100"
    onBlur={[Function]}
    onChange={[Function]}
    placeholder="Untitled"
    value=""
  />
  <h4>
    By Author A
  </h4>
</div>
`;

exports[`PollTitle (form) should match snapshot (default) 1`] = `
<div
  className="u-align-center"
>
  <input
    className="c-poll-display__input qa-title-input u-margin-half"
    maxLength="100"
    onBlur={[Function]}
    onChange={[Function]}
    placeholder="Untitled"
    value=""
  />
  <h4>
    By Anonymous
  </h4>
</div>
`;

exports[`PollTitle (view) should match snapshot (author) 1`] = `
<div
  className="u-align-center"
>
  <h1
    className="u-margin-half"
  >
    Untitled
  </h1>
  <h4>
    By Author A
  </h4>
</div>
`;

exports[`PollTitle (view) should match snapshot (default) 1`] = `
<div
  className="u-align-center"
>
  <h1
    className="u-margin-half"
  >
    Untitled
  </h1>
  <h4>
    By Anonymous
  </h4>
</div>
`;

exports[`PollTitle (view) should match snapshot (title) 1`] = `
<div
  className="u-align-center"
>
  <h1
    className="u-margin-half"
  >
    Title A
  </h1>
  <h4>
    By Anonymous
  </h4>
</div>
`;
